### DISCUSSION TITLE
Dependency Injection Using Splat

### DISCUSSION THREAD

**Discussion by aniljbhatia** [AvaloniaUI/Avalonia#7022] [General]
Does anybody know of a tutorial for using dependency injection using Splat?

I understand the principles and have most of the code in place but there is something I am missing. I haven't had that light bulb moment yet!

**Comment by maxkatz6** [comment]
I am not sure if Splat is classic choice for dependency injection, as well as I don't think it supports constructor/property injection.
Also, it's not related to Avalonia, we don't require any specific dependency injection library so you can choose any.
Splat is shipped together with ReactiveUI tho, but because it's used internally there.

**Comment by aniljbhatia** [comment]
It was a logical choice to use Splat. As you said it is part of ReactiveUI and the Avalonia example for routing uses it. It didn't make sense to add another library when one was already included.

I have been trying to find some articles for Avalonia and DI, but there doesn't seem to be much out there. Specifically around how to link everything together with the views and XAML. Are you aware of any?

As you can tell I am new to Avalonia and only used DI once many years ago, so any help would be much appreciated.

**Comment by danielmayost** [comment]
There is a nice article here:
https://dev.to/ingvarx/avaloniaui-dependency-injection-4aka

If you have any questions, I will try to help.
